iree/compiler/Dialect/HAL/Transforms/Passes.cpp

```diff
@@ -121,6 +121,12 @@ void buildHALTransformPassPipeline(OpPassManager &passManager,
     // if we serialized things.
     passManager.addPass(createSymbolDCEPass());
   }
+
+  // Run our own CSE on variable loads before moving on.
+  // When specifying side effects can help MLIR's core CSE pass eliminate
+  // redundant loads we can remove this.
+  passManager.addNestedPass<FuncOp>(createCanonicalizerPass());
+  passManager.addNestedPass<FuncOp>(createCSEVariableLoadsPass());
```

Comment:
move this up above the serialization stuff to be next to the standard CSE (and you don't need the extra canonicalizer pass) - that way when CSE does do the right thing it happens in the same place and we can just kill this pass